### PR TITLE
feat: Update wit version

### PR DESCRIPTION
**Pull request by liamwh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -439,7 +440,7 @@ We are looking for maintainers to add more parsers and to write query files for
 - [x] [wgsl](https://github.com/szebniok/tree-sitter-wgsl) (maintained by @szebniok)
 - [x] [wgsl_bevy](https://github.com/theHamsta/tree-sitter-wgsl-bevy) (maintained by @theHamsta)
 - [x] [wing](https://github.com/winglang/tree-sitter-wing) (maintained by @gshpychka, @MarkMcCulloh)
-- [x] [wit](https://github.com/liamwh/tree-sitter-wit) (maintained by @liamwh)
+- [x] [wit](https://github.com/liamwh/tree-sitter-wit) (experimental, maintained by @liamwh)
```
Comment (clason) [inline README.md:443]: Rebase error?

Comment (liamwh) [reply README.md]: Resolved, thx for pointing this out

File: README.md
Comment (ObserverOfTime) [inline README.md:1]: Don't edit the readme. The space doesn't matter and the extra line will be removed by CI.

File: queries/wit/highlights.scm [hunk 2]
```diff
@@ -1,78 +1,157 @@
...
+(comment) @comment
 
-(block_comment) @comment
+(ty
+  (id)) @type
+
+(package_decl
+  (id)) @module
+
+(valid_semver) @property
```
Comment (ObserverOfTime) [inline queries/wit/highlights.scm:9]:
```suggestion
(valid_semver) @string.special
```

File: queries/wit/highlights.scm [hunk 3]
```diff
@@ -1,78 +1,157 @@
...
+  name: (id) @type)
+
+(record_field
+  name: (id) @variable.member)
+
+(flags_items
+  name: (id) @type)
+
+(flags_body
+  (id) @property)
```
Comment (ObserverOfTime) [inline queries/wit/highlights.scm:59]:
```suggestion
  (id) @attribute)
```

**Comment by clason** [comment]
Please drop your changes to the readme entirely.

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/wit/highlights.scm (hunk 3 above)
Comment (liamwh) [reply queries/wit/highlights.scm:59]: Done ✅

File: README.md
Comment (liamwh) [reply README.md:1]: Done, removed README

File: queries/wit/highlights.scm [hunk 4]
```diff
@@ -1,78 +1,157 @@
...
-  "char"
-  "string"
-  ; Container Types
-  "list"
-  "tuple"
-  "option"
-  "result"
-] @type.builtin
+  "type"
+  "interface"
```
[the remark this answers is not in the document]
Comment (liamwh) [reply queries/wit/highlights.scm:92]: Agreed, done.

**Review by ObserverOfTime — APPROVED** [review]

File: queries/wit/highlights.scm [hunk 5]
```diff
@@ -1,78 +1,151 @@
-; Comments
-(line_comment) @comment
+(comment) @comment
```
Comment (ObserverOfTime) [inline queries/wit/highlights.scm:1]:
```suggestion
(comment) @comment @spell
```
Forgot this one.

File: queries/wit/highlights.scm [hunk 6]
```diff
@@ -1,78 +1,157 @@
...
-; Functions and Methods (lowercase names followed by parentheses)
-((identifier) @function
-  (#match? @function "^[a-z_][a-zA-Z0-9_]*%("))
+[
+  "tuple"
+  "list"
+  "option"
+  "result"
+  "borrow"
+] @type.definition
```
[the remark this answers is not in the document]
Comment (liamwh) [reply queries/wit/highlights.scm]: Done ✅

[the remark this answers is not in the document]
Comment (liamwh) [reply queries/wit/highlights.scm]: Done ✅

File: queries/wit/injections.scm [hunk 7]
```diff
@@ -1,5 +1,2 @@
-((line_comment) @comment
-  (#set! injection.language "comment"))
-
-((block_comment) @comment
+((comment) @comment
```
[the remark this answers is not in the document]
Comment (clason) [reply queries/wit/injections.scm:1]: @liamwh can you make that change? This is the last blocker, and I want to stop having to drop `wit` from the automated parser updates.

[the remark this answers is not in the document]
Comment (clason) [reply queries/wit/injections.scm:1]: (And I can't make that change since you did not make this PR from a feature branch 😞 )

**Review by clason — CHANGES_REQUESTED** [review]

File: queries/wit/highlights.scm [hunk 8]
```diff
@@ -1,81 +1,150 @@
-; Comments
-(line_comment) @comment
+(comment) @injection.content @spell
```
Comment (clason) [inline queries/wit/highlights.scm:1]: no, not here -- in the injections.scm!

File: queries/wit/injections.scm (hunk 7 above)
[the remark this answers is not in the document]
Comment (liamwh) [reply queries/wit/injections.scm:1]: I have created the change, but kept `@spell`, so the line now reads `(comment) @injection.content @spell`. Is this now correct?

File: queries/wit/highlights.scm [hunk 9]
```diff
@@ -1,78 +1,151 @@
...
 ] @keyword.import
 
-; Resource Keywords
-"static" @keyword.modifier
-
-; Named Types (Capitalized identifiers)
-((identifier) @type
-  (#match? @type "^[A-Z]"))
+(result
+  "result" @keyword.return)
```
[the remark this answers is not in the document]
Comment (liamwh) [reply queries/wit/highlights.scm]: I removed this as result is actually a builtin type.

File: queries/wit/injections.scm (hunk 7 above)
[the remark this answers is not in the document]
Comment (clason) [reply queries/wit/injections.scm:1]: Nope, sorry. This comment is on the injections, not the highlights.

[the remark this answers is not in the document]
Comment (liamwh) [reply queries/wit/injections.scm:1]: Ok thanks for that, it should be correct now

[the remark this answers is not in the document]
Comment (clason) [reply queries/wit/injections.scm:1]: Still no 😞

File: queries/wit/highlights.scm (hunk 8 above)
Comment (clason) [inline queries/wit/highlights.scm:1]:
```suggestion
(comment) @comment @spell
```

File: queries/wit/injections.scm (hunk 7 above)
Comment (clason) [inline queries/wit/injections.scm:1]:
```suggestion
((comment) @injection.content
```

**Review by clason — CHANGES_REQUESTED** [review]
Like so (and only so; pay attention to the different files).

**Comment by clason** [comment]
You know what? I'll take care of it.

**Comment by clason** [comment]
Superseded by #6527

**Pull request closed without merging**